.github/workflows/release.yaml

```diff
@@ -0,0 +1,38 @@
+name: Release to Edge
+
+on:
+  push:
+    branches: [ main ]
+  release:
+    types: [ published ]
+
+jobs:
+  check:
+    uses: ./.github/workflows/_quality_check.yaml
+    secrets: inherit
+
+  release:
+    needs: check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Initialize lxd  # This should dropped once it's implemented on charming-actions itself. https://github.com/canonical/charming-actions/issues/140
+        if: ${{ inputs.destructive-mode == false }}
+        run: |
+          sudo lxd init --auto
+          # The following is needed to fix network error when creating new base instance on lxd
+          sudo chmod a+wr /var/snap/lxd/common/lxd/unix.socket
+          lxc network set lxdbr0 ipv6.address none
+          sudo usermod -a -G lxd $USER
+          if [[ $(cat /etc/os-release | grep VERSION_CODENAME) == *"jammy"* ]]; then
+            sudo iptables -F FORWARD
+            sudo iptables -P FORWARD ACCEPT
+          fi
+      - name: Upload charm to charmhub
+        uses: canonical/charming-actions/upload-charm@2.4.0
```

Comment:
What is all in one?

Reply:
starting charmcraft 2.5, `charmcraft.yaml` is the only file we need: https://juju.is/docs/sdk/charmcraft-yaml

Reply:
We need to use pre-release of actions, since that include [this](https://github.com/canonical/charming-actions/pull/132) change. I will add it + comment why we are suing pre-release version. Thanks